### DISCUSSION TITLE
Modal Dialog moves layout - bad UX, bad DX but PrimeVue creators seems to be fine with it :(

### DISCUSSION THREAD

**Discussion by yureckey** [primefaces/community#82] [PrimeVue] [answered]
I'm sorry for this rant, but I'm thinking about a move from Quasar/Vuetify to PrimeVue (not the first time) and there is too much "drag" and too many rough edges that shouldn't be there for a library of such maturity.
This one, for example:
modal dialog popup on show will hide body scrollbar and on close will get it back - which leads to whole layout small shift, which is just bad UX. This "behavior" (I would call it just a bug, though) is known for years! https://github.com/primefaces/primevue/issues/882 (and number of questions in discord) and nobody (from the team) care about it, since the issue was closed. So, it's up to developer to spent his time on research and implementing plausible workaround (.p-overflow-hidden { overflow: auto; } ) - that's bad DX
I'm really trying to give PrimeVue a chance, I bought UI kit, Template, Blocks - but this kind of UX, DX drives me off.
Another example, three days ago I posted a question here in Discussions about label on button being off center - it is not answered still.
These are two very very basic things, I'm not asking about some sophisticated request, it should work properly out of the box, just like in Quasar/Vuetify.

**Comment by cagataycivici** [comment, marked as answer]
Hi there!

About the modal dialog moving the layout, if the scrollbar has a width in your system then it is likely to happen, I think many of the dev team members including me has their macs set to auto mode so it does not cause an issue however in other environments like windows, it is an issue for sure.

<img width="427" alt="Screen Shot 2023-09-22 at 10 05 57" src="https://github.com/primefaces/community/assets/686247/86fe153d-0e11-4917-8a2f-00744142be90">

We'll tackle this one for next week's update definitely so thank you for bringing it to our attention.

I'll also talk about the button alignment case with the team where our designers will also join.

I think we can get these two for next week. Please monitor this post for updates, cheers.

**Reply by yureckey** [reply]
Thank you Çağatay!

**Reply by yureckey** [reply]
I see there is now (3.35) added adjustment for hidden scrollbar, looks better, thanks!

**Reply by cagataycivici** [reply]
Glad to hear!
